File: Dataset/Recursion_prepared/Python/0203-remove-linked-list-elements_trace_5.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode(next=head)
        prev, curr = dummy, head
        
        while curr:
            nxt = curr.next
            
            if curr.val == val:
                prev.next = nxt
            else:
                prev = curr
            
            curr = nxt
        return dummy.next

if __name__ == "__main__":
    # Test case 5
    head = ListNode(1, ListNode(2, ListNode(2, ListNode(1, ListNode(2)))))
    val = 2
    solution = Solution()
    result = solution.removeElements(head, val)
    while result:
        print(result.val, end=" -> ")
        result = result.next
    print("None")

File: Dataset/Advanced/Test/Recursion_prepared/Java/0203-remove-linked-list-elements_trace_5.java
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode(0, head);
        ListNode prev = dummy, curr = head;
        while(curr != null) {
            ListNode nxt = curr.next;
            
            if(curr.val == val)
                prev.next = nxt;
            else
                prev = curr;
            
            curr = nxt;
        }
        return dummy.next;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        ListNode head = new ListNode(1, new ListNode(2, new ListNode(3, new ListNode(4, new ListNode(5, new ListNode(6, new ListNode(7)))))));
        solution.removeElements(head, 4);
    }
}

class ListNode {
    int val;
    ListNode next;
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}